metadata/src/main/java/org/apache/kafka/metadata/BrokerRegistration.java

```diff
@@ -192,7 +192,7 @@ private BrokerRegistration(
         this.isMigratingZkBroker = isMigratingZkBroker;
         directories = new ArrayList<>(directories);
         directories.sort(Uuid::compareTo);
-        this.directories = Collections.unmodifiableList(directories);
+        this.sortedDirectories = Collections.unmodifiableList(directories);
```

Comment:
Seems like we should copy the list and sort it here to make sure that it actually is sorted. It should be a short list (a dozen or so max) so not a big performance issue.

Reply:
We do that in the prior 2 lines

Reply:
ah, sorry, that is my bad. Please ignore this then :)